core/src/main/java/org/mule/runtime/core/internal/util/rx/RxUtils.java

```diff
@@ -200,8 +200,11 @@
 
     return doPropagateCompletion(upstream, downstream, transformer,
                                  new AtomicInteger(0), completer, errorForwarder,
-                                 () -> delayedExecutor.schedule(() -> completer.runOnce(), completionTimeoutMillis,
-                                                                MILLISECONDS));
+                                 () -> delayedExecutor.schedule(() -> {
+                                   LOGGER.warn("Calling completer after {} milliseconds",
```

Comment:
add some context to the logged line to know where it came from. Also, in the case of graceful shutdown timeout, there is already a WARN being logged and this one may be redundant.